### ISSUE TITLE
Node.js  Loaders Team Meeting 2023-06-20

### ISSUE THREAD

**Issue by mhdawson** [nodejs/loaders#146]
## Time

**UTC Tue 20-Jun-2023 18:00 (06:00 PM)**:

| Timezone      | Date/Time             |
|---------------|-----------------------|
| US / Pacific  | Tue 20-Jun-2023 11:00 (11:00 AM) |
| US / Mountain | Tue 20-Jun-2023 12:00 (12:00 PM) |
| US / Central  | Tue 20-Jun-2023 13:00 (01:00 PM) |
| US / Eastern  | Tue 20-Jun-2023 14:00 (02:00 PM) |
| EU / Western  | Tue 20-Jun-2023 19:00 (07:00 PM) |
| EU / Central  | Tue 20-Jun-2023 20:00 (08:00 PM) |
| EU / Eastern  | Tue 20-Jun-2023 21:00 (09:00 PM) |
| Moscow        | Tue 20-Jun-2023 21:00 (09:00 PM) |
| Chennai       | Tue 20-Jun-2023 23:30 (11:30 PM) |
| Hangzhou      | Wed 21-Jun-2023 02:00 (02:00 AM) |
| Tokyo         | Wed 21-Jun-2023 03:00 (03:00 AM) |
| Sydney        | Wed 21-Jun-2023 04:00 (04:00 AM) |

Or in your local time:
* https://www.timeanddate.com/worldclock/fixedtime.html?msg=Node.js+Foundation+Loaders%20Team+Meeting+2023-06-20&iso=20230620T1800
* or https://www.wolframalpha.com/input/?i=06PM+UTC%2C+Jun+20%2C+2023+in+local+time

## Links

* Minutes Google Doc: <https://docs.google.com/document/d/1SHBPJVM5-rjI3L85q4OE5ZO6yoCgjFBjaRDfxuFyNME/edit>

## Agenda

Extracted from **loaders-agenda** labelled issues and pull requests from the **nodejs org** prior to the meeting.

## Invited

* Loaders team: @nodejs/loaders

## Notes

The agenda comes from issues labelled with `loaders-agenda` across **all of the repositories in the nodejs org**. Please label any additional issues that should be on the agenda before the meeting starts.

## Joining the meeting

* link for participants: <https://zoom.us/j/97506450082>

* For those who just want to watch: <https://www.youtube.com/c/nodejs+foundation/live>

**Comment by JakobJingleheimer** [comment]
Possibly message port via register()

**Comment by giltayar** [comment]
Would love to be part of the "message port in register()" discussion, but unfortunately I am not available at the time of the meeting.

**Comment by JakobJingleheimer** [comment]
Let's postpone that discussion til the following meeting then?

**Comment by GeoffreyBooth** [comment]
> Would love to be part of the “message port in register()” discussion, but unfortunately I am not available at the time of the meeting.

This can also happen async. I don’t have any particular ideas for how to implement this, so if you have a design in mind please feel free to propose it in an issue in this repo. I think what we were generally aiming for is:

- Provide a way to access the communications port between the application and loaders threads; on the application thread side it could perhaps be returned by `register`, and on the loaders side? `import { messagePort } from 'node:module'`? Or just do the API on both sides? Or one of the “hooks” could be the handler for the port, like we have `export async function resolve` we could also have `export async function onMessage` or whatever?
- Once that lands, follow up with a PR to get rid of the `globalPreload` hook. Update the docs to recommend everyone use `--import` with `register`, and whatever would’ve been in a `globalPreload` sloppy script can now be in whatever code is run via `--import` (i.e. everything else that’s in the file that calls `register`). In this PR or as a follow-up, deprecate the `--experimental-loader` flag.

On a separate track, I’m eager to see https://github.com/nodejs/node/pull/47999 completed, as that finishes one of the three remaining to-do items in our “get to stable” milestone. @aduh95 where did you leave off with that, and is there anything in particular that you’d welcome help with on that one?

Closing because it seems like we’re not meeting this week.
